pyro/contrib/timeseries/gp.py

```diff
@@ -111,3 +117,142 @@ def forecast(self, targets, dts):
         filtering_state = self._filter(targets)
         predicted_mean, predicted_covar = self._forecast(dts, filtering_state)
         return torch.distributions.Normal(predicted_mean, predicted_covar.sqrt())
+
+
+class LinearlyCoupledMaternGP(TimeSeriesModel):
+    """
+    A time series model in which each output dimension is modeled as a linear combination
+    of shared univariate Gaussian Processes with Matern kernels.
+
+    In more detail, the generative process is:
+
+        :math:`y_i(t) = \\sum_j A_{ij} f_j(t) + \\epsilon_i`
+
+    The targets :math:`y_i` are assumed to be evenly spaced in time. Training and inference
+    are logarithmic in the length of the time series T.
+
+    :param float nu: The order of the Matern kernel; either 1.5 or 2.5.
+    :param float dt: The time spacing between neighboring observations of the time series.
+    :param int obs_dim: The dimension of the targets at each time step.
+    :param int num_gps: The number of independent GPs that are mixed to model the time series.
+        Typical values might be :math:`\\N_{\\rm gp} \\in [\\D_{\\rm obs} / 2, \\D_{\\rm obs}]`
+    """
+    def __init__(self, nu=1.5, dt=1.0, obs_dim=2, num_gps=1,
+                 log_length_scale_init=None, log_kernel_scale_init=None,
+                 log_obs_noise_scale_init=None):
+        self.nu = nu
+        self.dt = dt
+        assert obs_dim > 1, "If obs_dim==1 you should use IndependentMaternGP"
+        self.obs_dim = obs_dim
+        self.num_gps = num_gps
+
+        if log_obs_noise_scale_init is None:
+            log_obs_noise_scale_init = -2.0 * torch.ones(obs_dim)
+        assert log_obs_noise_scale_init.shape == (obs_dim,)
+
+        self.dt = dt
+        self.obs_dim = obs_dim
+        self.num_gps = num_gps
+
+        super(LinearlyCoupledMaternGP, self).__init__()
+
+        self.kernel = MaternKernel(nu=nu, num_gps=num_gps,
+                                   log_length_scale_init=log_length_scale_init,
+                                   log_kernel_scale_init=log_kernel_scale_init)
+        self.full_state_dim = num_gps * self.kernel.state_dim
+
+        self.log_obs_noise_scale = nn.Parameter(log_obs_noise_scale_init)
+        self.A = nn.Parameter(0.3 * torch.randn(self.num_gps, self.obs_dim))
+
+    def _get_obs_matrix(self):
+        # (num_gps, obs_dim) => (state_dim * num_gps, obs_dim)
+        return self.A.repeat_interleave(self.kernel.state_dim, dim=0) * \
+            torch.tensor([1.0] + [0.0] * (self.kernel.state_dim - 1),
+                         dtype=self.A.dtype, device=self.A.device).repeat(self.num_gps).unsqueeze(-1)
```

Comment:
FYI you can also use the `.new_tensor()` method, e.g. this is equivalent:
```py
self.A.new_tensor([1.0] + [0.0] * (self.kernel.state_dim - 1)
                  ).repeat(self.num_gps).unsqueeze(-1)
```

Reply:
done